### ISSUE TITLE
Music script not working on macOS Big Sur

### ISSUE THREAD

**Issue by nicknisi** [nicknisi/dotfiles#161]
The AppleScript/JavaScript script that fetches the current artist and track is throwing an error on Big Sur.

```
execution error: Error: Error: Can't get object. (-1728)
```

**Comment by jrock2004** [comment]
![image](https://user-images.githubusercontent.com/655716/99278592-3be4ed80-27fd-11eb-896b-7263edcb36bd.png)

I think I altered mine from yours a while back but not sure

```
let output = "";
if (Application("Music").running()) {
  const track = Application("Music").currentTrack,
    artist = track.artist(),
    title = track.name();

  output = `${title} - ${artist}`.substr(0, 50);
} else if (Application("Spotify").running()) {
  const track = Application("Spotify").currentTrack,
    artist = track.artist(),
    title = track.name();

  output = `${title} - ${artist}`.substr(0, 50);
} else if (Application("iTunes").isRunning()) {
  const track = Application("iTunes").currentTrack,
    artist = track.artist(),
    title = track.name();

  output = `${title} - ${artist}`.substr(0, 50);
}

output;
```

**Comment by nicknisi** [comment]
That's on Big Sur? I tried running the script you posted and I'm still getting the same error. Is your music playing through Apple Music or is it Spotify?

<img width="1136" alt="image" src="https://user-images.githubusercontent.com/293805/99392403-4658c300-28a1-11eb-93a1-40d387082d62.png">

**Comment by jrock2004** [comment]
I guess, what are you using? Music or Spotify? I was using Music for mine

**Comment by nicknisi** [comment]
I'm using Music, too.

**Comment by jrock2004** [comment]
OS: macOS Big Sur
Version: 11.0.1

Terminal: iTerm2
Build: 3.4.0

tmux Version: 3.1c

**Comment by nicknisi** [comment]
After playing with it some more it seems like it works for tracks coming from playlists and albums but it fails for Apple Music streaming, such as radio stations.

**Comment by jrock2004** [comment]
Yeah, you seem to be right. I am trying to remember how to debug these scripts

**Comment by nicknisi** [comment]
Spotify also might not be working.

**Comment by dyesiboy31** [comment]
Im using macOS catalina. Spotify not working.

**Comment by TheRojam** [comment]
guys, espcially @nicknisi , you've done and do agreat job on this script. i'll be happy to see it working again! (but can't help due i'm not a programmer)

**Comment by jrock2004** [comment]
@TheRojam I think its a Big Sur defect. Got to wait for the next update

**Comment by TheRojam** [comment]
idk whats happend... but apple music streaming works again w/  in my iterm2 on 11.1.0
![image](https://user-images.githubusercontent.com/1362428/103172951-bde22080-4857-11eb-8a44-264faa77c896.png)

**Comment by nicknisi** [comment]
Is it possible that's a file from your Apple Music library? I'm trying now with a radio station and still getting the same error on macOS 11.1

**Comment by nicknisi** [comment]
I think this has been fixed in macOS 11.3! 

![image](https://user-images.githubusercontent.com/293805/116639426-24726a80-a92e-11eb-8333-4a374da14cd5.png)

**Comment by jrock2004** [comment]
Yeah, seems to be working for me as well

**Comment by AhmedAhmed** [comment]
Guys, I know this issue is closed but the root cause of the issue is the fact that iTunes is no longer part of the new version of macOS and so it is crashing each time when it tries to look for iTunes. I simply removed the if condition for iTunes and it worked.

**Comment by nicknisi** [comment]
@AhmedAhmed great callout! I initially left it in there when it switched to Music to accommodate systems that might not have upgraded, but it's well past time to remove it. I have done so in 0d6210d.
